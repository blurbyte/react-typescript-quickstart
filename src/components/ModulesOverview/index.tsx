import * as React from 'react';

import SecondaryHeadline from '../SecondaryHeadline';
import Text from '../Text';
import Table from '../Table';

class ModulesOverview extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <section>
        <SecondaryHeadline>Comprehensive modules overview</SecondaryHeadline>
        <Text>There are a lot of interesting tech choices:</Text>
        <Table>
          <thead>
            <tr>
              <Table.HeadCell>Related modules</Table.HeadCell>
              <Table.HeadCell>Description</Table.HeadCell>
            </tr>
          </thead>
          <tbody>
            <Table.Row>
              <Table.Cell>react, react-dom</Table.Cell>
              <Table.Cell>Reusable components for web applications</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>typescript</Table.Cell>
              <Table.Cell>JavaScript but with types (and more)</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>fusebox</Table.Cell>
              <Table.Cell>Great bundler with simple configuration shipped with handy task manager</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>tslint</Table.Cell>
              <Table.Cell>Reports TypeScript and React syntax errors</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>styled-components</Table.Cell>
              <Table.Cell>Scoped components CSS styles, how awesome is it?</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>jest</Table.Cell>
              <Table.Cell>Automated test framework with built in assertion library</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>enzyme</Table.Cell>
              <Table.Cell>React testing utilities</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>prettier, lint-staged</Table.Cell>
              <Table.Cell>Enforces consistent coding style</Table.Cell>
            </Table.Row>
          </tbody>
        </Table>
      </section>
    );
  }
}

export default ModulesOverview;
