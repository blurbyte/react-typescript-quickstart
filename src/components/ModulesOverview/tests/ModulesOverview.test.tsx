import * as React from 'react';
import { shallow } from 'enzyme';

import ModulesOverview from '../index';

describe('ModulesOverview', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ModulesOverview />);
    expect(wrapper).toMatchSnapshot();
  });
});
