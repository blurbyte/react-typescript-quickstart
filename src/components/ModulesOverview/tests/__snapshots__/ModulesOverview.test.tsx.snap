// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModulesOverview renders correctly 1`] = `
<section>
  <styled.h2>
    Comprehensive modules overview
  </styled.h2>
  <styled.p>
    There are a lot of interesting tech choices:
  </styled.p>
  <Table>
    <thead>
      <tr>
        <Styled(styled.th)>
          Related modules
        </Styled(styled.th)>
        <Styled(styled.th)>
          Description
        </Styled(styled.th)>
      </tr>
    </thead>
    <tbody>
      <styled.tr>
        <styled.td>
          react, react-dom
        </styled.td>
        <styled.td>
          Reusable components for web applications
        </styled.td>
      </styled.tr>
    </tbody>
  </Table>
</section>
`;
